### ISSUE TITLE
Azimuthal Equidistant incorrect for non-polar and global extent

### ISSUE THREAD

**Issue by QuLogic** [SciTools/cartopy#568]
If you use `AzimuthalEquidistant` with non-polar central latitudes (`central_latitude != +/- 90`) and _global_ extent, the projection does not project correctly.

For example, `central_latitude=40, central_longitude=-100` should be something like the following:
![correct](http://www.globmaritime.com/images/stories/users/azimuthal_equidistant_map_of_the_world.gif)

When attempted with Cartopy, the result is:
![figure_1](https://cloud.githubusercontent.com/assets/302469/6312009/7c2b5074-b93a-11e4-9afc-86b39c38157a.png)

I have marked 3 points at (100W, 40N) (correct - green), at (60W, 60S) (probably correct - blue), and at (60E, 60S) (not correct - red). I haven't had a chance to look too deeply, but is there some sort of optimization that prevents the paths from going the "long way" around? Maybe some constraint to [0, 2pi) or something?

```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

proj = ccrs.AzimuthalEquidistant(central_latitude=40, central_longitude=-100)
ax = plt.axes(projection=proj)

ax.coastlines()
ax.gridlines()

ax.plot(-100, 40, 'go', transform=ccrs.Geodetic())
ax.plot(60, -60, 'ro', transform=ccrs.Geodetic())
ax.plot(-60, -60, 'bo', transform=ccrs.Geodetic())

# Global is as above
ax.set_global()
# Local is mostly okay, but you get wrapped around stuff underneath
#ax.set_extent([-140, -70, 10, 60])

plt.show()
```

**Comment by QuLogic** [comment]
Ah, it seems if you set a spherical globe, then things are projected better:
![figure_1](https://cloud.githubusercontent.com/assets/302469/6313395/ebd6ee66-b975-11e4-84c2-a8a825bca5e4.png)

Some searching indicates this may be a [bug in proj4](http://trac.osgeo.org/proj/ticket/246). I guess we should add a warning or document this somewhere?

**Comment by QuLogic** [comment]
Looks like [basemap does the same](https://github.com/matplotlib/basemap/blob/master/lib/mpl_toolkits/basemap/proj.py#L112).

**Comment by pelson** [comment]
Hmmm. Does look like a Proj4 bug. Would you mind feeding that back to proj4 and linking the issues?

**Comment by QuLogic** [comment]
The [proj4 issue](http://trac.osgeo.org/proj/ticket/246) already existed with a not-quite-complete patch. Hopefully it will be included soon.

**Comment by QuLogic** [comment]
This bug has been fixed in proj 4.9.2, but I'll leave it open because we're not sure if we want to bump the requirements just yet.

**Comment by pelson** [comment]
> because we're not sure if we want to bump the requirements just yet.

**Comment by QuLogic** [comment]
On the other hand, there are some requests to reduce the dependency from 4.9 to 4.8. Maybe a warning with older version like with Robinson?

**Comment by pelson** [comment]
> Maybe a warning with older version like with Robinson?

Very sensible. I'll put that in v0.14 milestone. Want to take it on?

**Comment by QuLogic** [comment]
This should have been closed by #757.
